src/components/Auth/RegisterCheckBox.js

```diff
@@ -0,0 +1,100 @@
+import React, { useState, useCallback } from 'react';
+import * as S from 'components/Auth/styles';
+import styled from 'styled-components';
+
+const CheckDotLabel = styled.label`
+  display: inline-block;
+  vertical-align: middle;
+  align-items: center;
+  cursor: pointer;
+`;
+
+const CheckDot = styled.input.attrs({ type: 'checkbox' })`
+  display: none;
+  overflow: hidden;
+  padding: 0;
+  position: absolute;
+  white-space: nowrap;
+`;
+
+const CustomCheckDot = styled.div`
+  display: inline-block;
+  width: 13px;
+  height: 13px;
+  border-radius: 50%;
+  margin-right: 20px;
+  border: solid 1px #c4c4c4;
+  transition: all 150ms;
+  ${CheckDot}:checked + & {
+    border: solid 1px #ffd25d;
+    background-color: #ffd25d;
+  }
+`;
+
+const RegisterCheckBox = ({ checked, ...props }) => {
+  const [checkedList, setCheckedLists] = useState([]);
+  const dataLists = [
+    { id: 1, text: '개인정보 이용약관(필수)' },
+    { id: 2, text: 'koala 이용약관(필수)' },
+  ];
+
+  const onCheckedAll = useCallback(
+    (checked) => {
+      if (checked) {
+        const checkedListArray = [];
+        dataLists.forEach((list) => checkedListArray.push(list.id));
+        setCheckedLists(checkedListArray);
```

Comment:
흠... 제가 봤을 땐 굳이 forEach와 push 를 중첩해서 복잡하게 사용하지 않아도 괜찮아 보입니다.
제 착오가 있었다면 수정하지 않으셔도 됩니다.

```suggestion
        setCheckedLists(dataLists.map(({id}) => id));
```

Reply:
흠... 이걸 보니 forEach와 map의 차이를 좀 더 확실히 알고 수정해보겠습니다.

Reply:
reduce 도 알아놓으면 자주 씁니다~

Reply:
이 둘은 배열의 요소를 반복한다는 공통점이 있어서 둘이 코드를 작성했을 때 실행에 대해서는 정상작동합니다.
차이점으로는 forEach는 각 요소에 대해 콜백을 실행하며 값을 반환하지 않고,
map은 각 요소에서 함수를 호출하여 결과로 새 배열을 작성하여 각 요소를 새 요소에 매핑하다는 것입니다.
둘 다 차이가 그렇게 없고 코드의 양과 반복의 결과에 대한 조작이 필요하기에 map으로 수정하겠습니다.

Reply:
8d15007